### PR TITLE
Handle network manager default in openSUSE

### PR DESCRIPTION
**Pull request by jknphy**
Handle network manager default in openSUSE
- Related ticket: https://progress.opensuse.org/issues/40067
- Needles: [needles opensuse](https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/437)
- Verification run: 
  - [Tumbleweed-yast2_ncurses](http://dhcp87.suse.cz/tests/2731)
  - [Tumbleweed-yast2_gui](http://dhcp87.suse.cz/tests/2730)
  - [Tumbleweed-extra_tests_on_gnome](http://dhcp87.suse.cz/tests/2729)

### CODE REVIEW

File: tests/yast2_gui/yast2_network_settings.pm [hunk 1]
```diff
@@ -17,66 +17,76 @@
...
+    $self->launch_yast2_module_x11('lan', target_match => [qw(yast2-lan-ui yast2_still_susefirewall2)], match_timeout => 60);
+    if (match_has_tag 'yast2_still_susefirewall2') {
         record_soft_failure "bsc#1059569";
         send_key 'alt-i';
         wait_still_screen;
     }
-
-    #	Global Options
-    send_key 'alt-g';
+    if (is_leap('>=15.1')) {
```
Comment (rwx788) [inline tests/yast2_gui/yast2_network_settings.pm]: Could you please add a comment that Network manager is now by default on leap?

File: tests/yast2_gui/yast2_network_settings.pm [hunk 2]
```diff
@@ -17,66 +17,76 @@
...
+    $cmd{routing}         = 'alt-u';
+
+    $self->launch_yast2_module_x11('lan', target_match => [qw(yast2-lan-ui yast2_still_susefirewall2)], match_timeout => 60);
+    if (match_has_tag 'yast2_still_susefirewall2') {
         record_soft_failure "bsc#1059569";
         send_key 'alt-i';
         wait_still_screen;
     }
-
-    #	Global Options
```
Comment (rwx788) [inline tests/yast2_gui/yast2_network_settings.pm]: Could you please also put new or recover old comment, I believe they are helpful even if you have introduced better named variables.

File: tests/yast2_gui/yast2_network_settings.pm [hunk 3]
```diff
@@ -17,66 +17,80 @@
...
+    $self->launch_yast2_module_x11('lan', target_match => [qw(yast2-lan-ui yast2_still_susefirewall2 yast2-lan-warning-network-manager)], match_timeout => 60);
+    if (match_has_tag 'yast2_still_susefirewall2') {
         record_soft_failure "bsc#1059569";
         send_key 'alt-i';
         wait_still_screen;
     }
-
-    #	Global Options
-    send_key 'alt-g';
+    elsif (match_has_tag 'yast2-lan-warning-network-manager' || is_leap('>=15.1')) {
```
Comment (okurz) [inline tests/yast2_gui/yast2_network_settings.pm]: I recommend to not use the leap version here but rather a function to determine if networkmanager is the default or wicked. Within this function you can simply state that openSUSE has network manager as default except for older leap versions. Leap should never have any "newer" feature which Tumbleweed does not have

File: tests/yast2_gui/yast2_network_settings.pm (hunk 2 above)
Comment (jknphy) [reply tests/yast2_gui/yast2_network_settings.pm]: yes, it is true that for separating the tab makes sense, I wrote new ones, but definitively there are many useless in these tests in general.

File: lib/y2_common.pm [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+our @EXPORT = qw(is_network_manager_default
+  continue_info_network_manager_default
+  accept_warning_network_manager_default
+);
+
+=head2 is_network_manager_default
+openSUSE has network manager as default except for older leap versions
+=cut
+sub is_network_manager_default {
+    return is_opensuse unless is_leap('<=15.0');
```
Comment (rwx788) [inline lib/y2_common.pm]: I know that this peace of code works, but just to be consistent, don't you mind rewriting it as 
`return is_opensuse && !is_leap('<=15.0')`. Problem with consistency here is that we explicitly return 0 or 1 if it's not openSUSE, and undef if it is. 0 and undef both mean false, but we can be consistent here.
It's a neat pick, so feel free to leave it as it is ;)

Comment (rwx788) [reply lib/y2_common.pm]: To be honest for me it's less readable, but that's my personal feeling about it ;)

**Pull request merged**